src/ripple/app/ledger/InboundLedger.cpp

```diff
@@ -353,9 +353,9 @@ void InboundLedger::done ()
     {
         mLedger->setClosed ();
         mLedger->setImmutable ();
-
         if (mReason != fcHISTORY)
             getApp().getLedgerMaster ().storeLedger (mLedger);
+        getApp().getInboundLedgers().onLedgerFetched(mReason);
```

Comment:
It's true that inside `onLedgerFetched` we return if `mReason != fcHISTORY`. But we want the callback to be generic and always run.